owa/app/js/components/manage-autogeneration-option/ManageAutogenerationOption.jsx

```diff
@@ -0,0 +1,455 @@
+                          </FormGroup>
+                          <FormGroup>
+                            <Label>Source to auto generate from: </Label>
+                            <Input
+                              onChange={this.onChange}
+                              type="select"
+                              defaultValue={currentGenOption.autogenerateSource}
+                              name="autogenerateSource"
+                            >
+                              <option value="">Select a source</option>
+                              {AllAutoGenerateSources}
+                            </Input>
+                          </FormGroup>
+                          <FormGroup>
+                            <Label>Options: </Label>
+                          </FormGroup>
+                          <div className="optionsCont">
+                            <FormGroup>
+                              <Label check>
+                                <Input
+                                  onClick={this.onChange}
+                                  name="autoGenerationEnabled"
+                                  checked={currentGenOption.autoGenerationEnabled}
+                                  type="checkbox"
+                                /> 
+                                Automatic Generation Enabled? 
+                              </Label>
+                            </FormGroup>
+                            <FormGroup>
+                              <Label check>
+                                <Input
+                                  onClick={this.onChange}
+                                  name="manualEntryEnabled"
+                                  checked={currentGenOption.manualEntryEnabled}
+                                  type="checkbox"
+                                /> 
+                                Manual Entry Enabled?
+                              </Label>
+                            </FormGroup>
+                          </div>
+                        </div>
+                      )}
+                      
+                    </ModalBody>
+                    <ModalFooter>
+                      { !firstLevelCompleted ? (
+                        <Button
+                          color="primary"
+                          onClick={this.nextStage}
+                          className="color-green"
+                        >
+                          <span>Next</span>
+                        </Button>
+                      ) : (
+                        <Button
+                          color="primary"
+                          className="color-green"
+                          onClick={saveGenOptionStatus === 'new' ? this.saveNewAutoGenOptions :
+                            this.editCurrentGenOption}
+                        >
+                          <span>Save</span>
+                        </Button>
+                      ) }
+                      {' '}
```

Comment:
An HTML tag like `span` being used to render space would be more appropriate here, rendering `{' '}` is not very helpful when reading this code. I may be wrong. :-)

Reply:
You are right. I'll change that. Thanks.